crates/bevy_ecs/src/core/entity_builder.rs

```diff
@@ -121,7 +121,10 @@ impl EntityBuilder {
             } else {
                 max_align as *mut _
             };
-            for (ty, offset) in self.info.drain(..) {
+            for ty in self.info.drain(..) {
+                let offset = *self.offsets
+                    .get(&ty.id())
+                    .unwrap();
```

Comment:
Perfectly, this should be optimized too, but for now I have no idea how to simplify it. Will think a bit later on it.